src/main/java/io/cryostat/discovery/CustomDiscovery.java

```diff
@@ -95,6 +99,21 @@ void onStart(@Observes StartupEvent evt) {
     @RolesAllowed("write")
     public Response create(
             Target target, @RestQuery boolean dryrun, @RestQuery boolean storeCredentials) {
+        try {
+            URIRange range = URIRange.fromString(uriRangeSetting);
+            if (!range.validate(target.connectUrl)) {
```

Comment:
How well does this work when the target connectUrl is a JMX URL (not an HTTP(S) one)? Could you add some unit test cases for this?